lib/ElementHandle.js

```diff
@@ -103,6 +103,22 @@ class ElementHandle {
     const objectId = this._remoteObject.objectId;
     return this._client.send('DOM.setFileInputFiles', { objectId, files });
   }
+
+  /**
+   * @return {!Promise<Object>}
+   */
+  async boundingBox() {
+    const layoutMetrics = await this._client.send('Page.getLayoutMetrics');
+    const boxModel = await this._client.send('DOM.getBoxModel', { objectId: this._remoteObject.objectId });
```

Comment:
Let's not use `DOM.getBoxModel` - it misbehaves with page zoom. It's also an experimental protocol method (and we try to minimize our use of experimental methods so that it would be easier to stabilize them in future).

The logic here is implementable with DOM API, e.g. [like this](https://cs.chromium.org/chromium/src/third_party/WebKit/Source/devtools/front_end/dom_extension/DOMExtension.js?q=DOMExtension&sq=package:chromium&l=494). Do you have any concerns about this approach?

Reply:
> Using a while loop to get the parent frame for offsets might not work because of CORS policy which makes the method inconsistant.

Yes, you're right.

We're thinking about introducing the `elementHandle.ownerFrame()` method in #433. This would allow you to iterate frame structure avoiding CORS and use bullet-proof `element.getBoundingClientRect` instead of `DOM.getBoxModel`.

If this solves your concerns, I'd like to wait for the `ownerFrame` to land first and then follow-up with this PR.

Reply:
OK, will do.